starlite/config/cors.py

```diff
@@ -12,33 +16,155 @@ class CORSConfig(BaseModel):
+                origins.append(self.allow_origin_regex)
+            setattr(
+                self, "_compiled_regex", re.compile("|".join([origin.replace("*.", r".*\.") for origin in origins]))
+            )
+        return cast("Pattern", self._compiled_regex)
+
+    @property
+    def is_allow_all_origins(self) -> bool:
+        """Get a cached boolean flag dictating whether all origins are allowed.
+
+        Returns:
+            Boolean dictating whether all origins are allowed.
+        """
+        if self._is_allow_all_origins is Empty:
+            setattr(self, "_is_allow_all_origins", "*" in self.allow_origins)
+        return cast("bool", self._is_allow_all_origins)
+
+    @property
+    def is_allow_all_methods(self) -> bool:
+        """Get a cached boolean flag dictating whether all methods are allowed.
+
+        Returns:
+            Boolean dictating whether all methods are allowed.
+        """
+        if self._is_allow_all_methods is Empty:
+            setattr(self, "_is_allow_all_methods", "*" in self.allow_methods)
+        return cast("bool", self._is_allow_all_methods)
+
+    @property
+    def is_allow_all_headers(self) -> bool:
+        """Get a cached boolean flag dictating whether all headers are allowed.
+
+        Returns:
+            Boolean dictating whether all headers are allowed.
+        """
+        if self._is_allow_all_headers is Empty:
+            setattr(self, "_is_allow_all_headers", "*" in self.allow_headers)
+        return cast("bool", self._is_allow_all_headers)
+
+    @property
+    def preflight_headers(self) -> Dict[str, str]:
+        """Get cached pre-flight headers.
+
+        Returns:
+            A dictionary of headers to set on the response object.
+        """
+        if self._preflight_headers is Empty:
+            headers: Dict[str, str] = {"Access-Control-Max-Age": str(self.max_age)}
+            if self.is_allow_all_origins:
+                headers["Access-Control-Allow-Origin"] = "*"
+            else:
+                headers["Vary"] = "Origin"
+            if self.allow_credentials:
+                headers["Access-Control-Allow-Credentials"] = str(self.allow_credentials).lower()
+            if not self.is_allow_all_headers:
+                headers["Access-Control-Allow-Headers"] = ", ".join(
+                    sorted(set(self.allow_headers) | DEFAULT_ALLOWED_CORS_HEADERS)
+                )
+            if self.allow_methods:
+                headers["Access-Control-Allow-Methods"] = ", ".join(
+                    sorted(
+                        set(self.allow_methods)
+                        if not self.is_allow_all_methods
+                        else {"DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"}
+                    )
+                )
+            setattr(self, "_preflight_headers", headers)
+
+        return cast("Dict[str, str]", self._preflight_headers)
+
+    @property
+    def simply_headers(self) -> Dict[str, str]:
```

Comment:
```suggestion
    def simple_headers(self) -> Dict[str, str]:
```

Reply:
Typo, I guess, because you store computed value of this in `self._simple_headers`